core/devmode/src/main/java/io/quarkus/dev/DevModeMain.java

```diff
@@ -54,6 +58,9 @@ public static void main(String... args) throws Exception {
         classesRoot = new File(args[0]);
         wiringDir = new File(args[1]);
         cacheDir = new File(args[2]);
+        group = args[3];
```

Comment:
I feel like we shouldn't be passing this in this way; the input should simply be the build configuration.

Reply:
I agree that this is not cool, but I am not clear on how to pass the build configuration to the runtime runner

Reply:
The "normal" approach would be to rely 100% on the configuration value and use a config root whose visibility is `BUILD_AND_RUN_TIME_FIXED`.  However as discussed above, this wouldn't suffice in the case where no value was given and the default has to be computed.

So, we have to bring it in some other way.  There are a few possibilities off the top of my head:

* Produce a resource that contains the application name & version and load it at boot using a static init build step
* Put it in a MANIFEST.MF property and load it at boot using a static init build step
* Just generate a class with fields containing the name and version, hard-coded to the computed name and version

Reply:
This is exactly the info I needed to close the loop, thank you! Now everything makes sense. I hadn't seen the any relevant parts of the code so I didn't know how to handle such cases, but with the explanation you just provided plus the information from the other comment I see the way forward.

I'll pick this up again tomorrow most likely and come up with proper solution that adheres to these guidelines.

Thanks for taking the time to explain!